internal/yunikorn/rest_test.go

```diff
@@ -126,6 +127,616 @@ func TestRESTClient_GetApplication(t *testing.T) {
+		},
+		{
+			name: "Server Error",
+			setup: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					http.Error(w, "server error", http.StatusInternalServerError)
+				}))
+			},
+			expected: nil,
+			wantErr:  true,
+		},
+		{
+			name: "Unexpected JSON",
+			setup: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					type unexpected struct {
+						Unexpected string `json:"unexpected"`
+					}
+					err := json.NewEncoder(w).Encode(&unexpected{Unexpected: "unexpected"})
+					if err != nil {
+						http.Error(w, err.Error(), http.StatusInternalServerError)
+					}
+				}))
+			},
+			expected: nil,
+			wantErr:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts := tt.setup()
+			defer ts.Close()
+
+			client := NewRESTClient(getMockServerYunikornConfig(t, ts.URL))
+
+			partitions, err := client.GetPartitions(context.Background())
+			if tt.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, tt.expected, partitions)
+			}
+		})
+	}
+}
+
+func TestRESTClient_GetPartitionQueues(t *testing.T) {
+	tests := []struct {
+		name     string
+		setup    func() *httptest.Server
+		expected *dao.PartitionQueueDAOInfo
+		wantErr  bool
+	}{
+		{
+			name: "200 OK Response",
+			setup: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					response := dao.PartitionQueueDAOInfo{
+
+						QueueName: "queue1",
+					}
+					err := json.NewEncoder(w).Encode(response)
+					if err != nil {
+						http.Error(w, err.Error(), http.StatusInternalServerError)
+					}
+				}))
+			},
+			expected: &dao.PartitionQueueDAOInfo{
+				QueueName: "queue1",
+			},
+
+			wantErr: false,
+		},
+		{
+			name: "Server Error",
+			setup: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					http.Error(w, "server error", http.StatusInternalServerError)
+				}))
+			},
+			expected: nil,
+			wantErr:  true,
+		},
+		{
+			name: "Unexpected JSON",
+			setup: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					type unexpected struct {
+						Unexpected string `json:"unexpected"`
+					}
+					err := json.NewEncoder(w).Encode(&[]unexpected{{Unexpected: "unexpected"}})
+					if err != nil {
+						http.Error(w, err.Error(), http.StatusInternalServerError)
+					}
+				}))
+			},
+			expected: nil,
+			wantErr:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts := tt.setup()
+			defer ts.Close()
+
+			client := NewRESTClient(getMockServerYunikornConfig(t, ts.URL))
+
+			queues, err := client.GetPartitionQueues(context.Background(), "testPartition")
+			if tt.wantErr {
+				require.Error(t, err)
```

Comment:
Can you also assert the error message is what we expect?

Reply:
added